Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs

```diff
@@ -87,4 +94,47 @@ private static string GetPermissions()
 
         return Encoding.UTF8.GetString(stream.ToArray());
     }
+
+    public async Task Update(OAuthClient client, CancellationToken cancellationToken)
+    {
+        var cachedApplication = _cachedApplications.FirstOrDefault(ca => ca.ClientId == client.ClientId);
+        if (cachedApplication != null)
+        {
+            cachedApplication.DefaultTier = client.DefaultTier;
+            await _applicationManager.UpdateAsync(cachedApplication, cancellationToken);
+            return;
+        }
+
+        var application = await _applicationManager.FindByIdAsync(client.ClientId, cancellationToken) ?? throw new NotFoundException(nameof(OAuthClient));
+        application.DefaultTier = client.DefaultTier;
+        await _applicationManager.UpdateAsync(application, cancellationToken);
```

Comment:
Don't you notice when you're duplicating code??

```suggestion
        var application = _cachedApplications.FirstOrDefault(ca => ca.ClientId == client.ClientId);
        if (application == null)
        {
            application = await _applicationManager.FindByIdAsync(client.ClientId, cancellationToken)?? throw new NotFoundException(nameof(OAuthClient));
        }

        application.DefaultTier = client.DefaultTier;
        await _applicationManager.UpdateAsync(application, cancellationToken);
```